### ISSUE TITLE
Class Inheritance Issues

### ISSUE THREAD

**Issue by richterd** [frankarendpoth/frankarendpoth.github.io#5]
Hi, 
thank you for the Tutorial on the Platformer Game (https://github.com/frankarendpoth/frankarendpoth.github.io/tree/master/content/pop-vlog/javascript/2018/006-rabbit-trap). I really enjoyed watching it. While implementing an own version of it, I recognized an issue with how you implement the "class inheritance".

The way you do it is the following:
```javascript
Superclass = function(){
    //Definition of Superclass attributes
}
Superclass.prototype = {
    //Definition of Superclass methods
}
Subclass = function(){
    Superclass.call(this);
    //Definition of Subclass attributes
}
Subclass.prototype = {
    //Definition of Subclass methods
}
Object.assign(Subclass.prototype, Superclass.prototype);
Subclass.prototype.constructor = Subclass;
```
This seems to work for your approach, but as soon as you want to override methods, your implementation leads to always calling the superclass' method instead of the subclass one (Similar to the constructor, that you also had to fix after the assignment).

A nicer way of implementing your approach, and also have the correct overriding behavior would be to turn around the assignment and e.g. use the following instead:
```javascript
Superclass = function(){
    //Definition of Superclass attributes
}
Superclass.prototype = {
    //Definition of Superclass methods
}
Subclass = function(){
    Superclass.call(this);
    //Definition of Subclass attributes
}
Subclass.prototype = Object.assign(Object.create(Superclass.prototype), {
    //Definition of Subclass methods
});
```
That way, the subclass' prototype methods correctly overwrite the Superclass once. Method overriding works as intended. No need to fix constructors.

Best,
Daniel

**Comment by frankarendpoth** [comment]
Yeah, my inheritance method is not well suited to overrides. Also,
technically, since the sub class recreates the prototype of the super class
it isn't true prototype inheritance if you really think about it, just
prototype cloning. I generally have to define any method overrides after
defining the prototype by doing: subclass.prototype.inheritedMethod =
function () {}; Which is basically just replacing the method I copied from
the super's prototype. I really like your implementation, though. What is
that syntax you use to define the subclass's prototype at the end using a
comma? I will test it out when I have some time.

**Comment by frankarendpoth** [comment]
I'm sorry, I'm reading this on a phone and I misread your code. Yours is
much cleaner and I will definitely test it out. I will probably make a
correction video as well. Excellent catch!
